packages/ag-charts-enterprise/src/features/annotations/axisButton.ts

```diff
@@ -0,0 +1,134 @@
+import { _ModuleSupport, _Scene } from 'ag-charts-community';
+
+import type { Coords } from './annotationTypes';
+
+const { BaseModuleInstance, Validate, BOOLEAN, createElement, REGIONS, ChartAxisDirection } = _ModuleSupport;
+
+export const DEFAULT_ANNOTATION_AXIS_BUTTON_CLASS = `annotations-axis-button`;
+
+export class AxisButton extends BaseModuleInstance implements _ModuleSupport.ModuleInstance {
+    @Validate(BOOLEAN)
+    public enabled = true;
+
+    private readonly button: HTMLButtonElement;
+    private readonly wrapper: HTMLElement;
+    private coords?: Coords;
+
+    setup(domManager: _ModuleSupport.DOMManager) {
+        const wrapper = domManager.addChild(
+            'canvas-overlay',
+            `${DEFAULT_ANNOTATION_AXIS_BUTTON_CLASS}-${this.axisCtx.axisId}`
+        );
+        wrapper.classList.add(DEFAULT_ANNOTATION_AXIS_BUTTON_CLASS);
+
+        const button = createElement('button');
+        wrapper.appendChild(button);
+
+        return {
+            wrapper,
+            button,
+        };
+    }
+
+    destroyElements(domManager: _ModuleSupport.DOMManager) {
+        domManager.removeChild('canvas-overlay', DEFAULT_ANNOTATION_AXIS_BUTTON_CLASS);
+    }
```

Comment:
Nit: Make these `setup()` & `destroyElements` methods:
- `private`
- Use `this.ctx.domManager` rather than passing it in as a parameter?
- Appear immediately AFTER the constructor?

Reply:
+1

I find a good ordering on class functions is: static fns -> constructor -> public fns -> private fns in the order they are referenced by constructor and other fns